src/methods/defining_methods/hard/q4/MainTest.java

```diff
@@ -0,0 +1,71 @@
+package methods.defining_methods.hard.q4;
+
+import java.util.stream.Stream;
+
+import global.MethodTest;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import global.BaseTest;
+import global.tools.CustomAssertions;
+import global.tools.TestOption;
+import global.variables.Clause;
+import global.variables.clauses.IntegerLiteral;
+import global.variables.clauses.StringLiteral;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class MainTest extends BaseTest {
+    // Java
+
+    public Clause[] testSentence() {
+        TestOption.isInputTest = true;
+        TestOption.defaultInput = "5 10";
+        return new Clause[]{
+                new StringLiteral("Enter the first number: "),
+                new StringLiteral("Enter the second number: "),
+                new StringLiteral("The greater number is "),
+                new IntegerLiteral("number")
+        };
+    }
+
+    public void runMain() {
+        GreaterThan.main(new String[0]);
+    }
+
+    static Stream<Arguments> correctInputProvider() {
+        return Stream.of(
+                Arguments.of(1, 2, 2),
+                Arguments.of(2, 1, 2),
+                Arguments.of(-1, -2, -1),
```

Comment:
Should have some test cases where some of the integers are farther apart as well.